in-progress/8754-slashing-and staking.md

```diff
@@ -0,0 +1,357 @@
+
+and gossips through to the validator set. 
+
+Validators who receive this proposal first also read the L1 for the list of `TxHash` posted along the block in question. If it does not match what's in the slashing proposal, it discards the slashing proposal. 
+
+If the list of `TxHash` does match, they request all `TxEffects` and proofs from the p2p. As a last resort, they request from the validator who initiated the slashing proposal. They also read the state root posted to L1. 
+
+If they're able to retreive all the required data to re-xecute the block contents and find an invalid state root, they sign the slashing proposal and gossip the signed proposal via the p2p. 
+
+The executing validator who initialized the slashing proposal must aggregate signatures and post to L1 for verification once >50% of the stake has signed the proposal. 
+
+On the L1, the contract verifies that: 
+1) The epoch has indeed re-orged.
+2) The committee members named in the slashing proposal have signed the specific block in question.
+3) Validators in control of more than 50% of stake have signed the slashing proposal.
+
+If the above checks pass, the named committee members are slashed. The L1 contracts pays out a small fee to cover the cost of verifying signatures on the L1. 
+
+```mermaid
+sequenceDiagram
+title Slashing an invalid state root - Happy Path
+
+    participant A as Proposer
+    participant B as Committee
+    participant C as Validator Set
+    participant D as Executing Validator
+    participant E as Ethereum L1
+
+    par
+        A ->> B: Gossip proposal
+        A ->> C: Gossip proposal
+    end
+
+    par
+        B ->> A: Gossip Attestations
+        B ->> C: Gossip Attestations
+    end
+
+    A ->> E: Propose Block
+    D ->> E: Retrieve TxHash
+    par
+        D ->> B: Request TxEffect + Proofs
+        D ->> C: Request TxEffect + Proofs
+    end
+
+    D ->> D: Execute block and check state root on L1
+    
+    alt If state root matches L1
+        D ->> D: Do nothing
+    else If state root does not match L1
+        D ->> C: Send proposal to slash malicious signatures
+    end
+
+    C ->> E: Retrieve Tx Hash
+    C ->> D: Request TxEffect + Proofs
+    C ->> C: Execute Proposal
+    alt If state root matches L1
+        C ->> C: Do nothing
+    else If state root doesn't match L1
+        C ->> D: Sign slashing proposal
+    end
+    D ->> D: Aggregate BLS signatures
+    D ->> E: Send BLS signatures and aggregated public key
+```
+
+**What should the validators do if they can't obtain the necessary data to re-execute the invalid state root?** 
+
+The above scenario describes the "happy path" where a stupid committee coordinates an attack and shares the data with the rest of the validator set. In a sophisticated attack, the committee will plot to withhold the data. 
+
+The question boils down to: Do we punish data withholding attacks given that they cannot be verified by the L1? 
+
```

Comment:
The mechanism for producing a data witholding attack claim will look different to this. Im not sure there is a way for a validator to make a claim like this without some kind of DAS implementation ( that i am not intimately familiar with ). 

I imagine data witholding slashing will make its way to apella rather than being sorted among the validators themselves